default/python/universe_generation/options.py

```diff
@@ -33,29 +33,30 @@
 # The following two options are used to determine the minimum number of systems and planets. This limit is
 # HS_MIN_SYSTEMS_IN_VICINITY systems and HS_MIN_PLANETS_IN_VICINITY_PER_SYSTEM planets per system within the near
 # vicinity of a home system, capped at HS_MIN_PLANETS_IN_VICINITY_TOTAL
-HS_MIN_SYSTEMS_IN_VICINITY = 8
-HS_MIN_PLANETS_IN_VICINITY_TOTAL = 10
-HS_MIN_PLANETS_IN_VICINITY_PER_SYSTEM = 1
+# HS_MIN_SYSTEMS_IN_VICINITY = 12
+# HS_MIN_PLANETS_IN_VICINITY_TOTAL = 12
+# HS_MIN_PLANETS_IN_VICINITY_PER_SYSTEM = 1
 
 # This option defines the extend of what is considered the "near vicinity" of a home system. This are all systems that
 # are within the number of jumps specified by HS_VICINITY_RANGE.
-HS_VICINITY_RANGE = 3
+HS_MIN_VICINITY_RANGE = 2
+HS_MAX_VICINITY_RANGE = 7
 
 # This options sets the maximum starting value for the minimum jump distance limit required between home systems.
 # With large galaxies an excessive amount of time can be used in failed attempts to select home systems, so defining
 # an upper limit for the home system selection process to use when calculating the starting value for the minimum
 # jump distance limit is reasonable.
-HS_MAX_JUMP_DISTANCE_LIMIT = 10
+HS_MAX_JUMP_DISTANCE_LIMIT = 20
 
 # This options defines the minimum jump distance limit between home systems that should be considered high priority.
 # As long as the jump distance limit which home systems must at least be apart does not get reduced below this limit
 # during the home system selection process, the minimum systems in home system vicinity requirement takes
 # precedence over the jump distance limit. If the jump distance limit drops below this minimum jump distance limit,
 # the process is restarted giving the jump distance limit precedence.
-HS_MIN_DISTANCE_PRIORITY_LIMIT = 5
+HS_MIN_DISTANCE_PRIORITY_LIMIT = 4
 
 # These two options define which types of planets are counted when determining the number of planets in the near
 # vicinity of a home system. HS_ACCEPTABLE_PLANET_SIZES is actually only needed for the process of adding planets
 # to the near vicinity of a home system in case that's needed to meet the limit.
-HS_ACCEPTABLE_PLANET_TYPES = planet_types_real + (fo.planetType.asteroids,)
-HS_ACCEPTABLE_PLANET_SIZES = planet_sizes_real + (fo.planetSize.asteroids,)
+HS_ACCEPTABLE_PLANET_TYPES = planet_types_real + (fo.planetType.asteroids, fo.planetType.gasGiant,)
+HS_ACCEPTABLE_PLANET_SIZES = planet_sizes_real + (fo.planetSize.asteroids, fo.planetSize.gasGiant,)
```

Comment:
This is potentially very problematic. Although I understand why you want to include them, so that players starting with Sly also get covered, this will lead to issues for everyone else.

Gas giants have been excluded for a reason, because they are very common, and if you count them as eligible vicinity planets for the evaluation of starting positions, all empires except those starting with Sly will most likely end up with far worse starting conditions than before.

When the original algorith has been implemented, gas giants haven't been valid colonization targets for any race. That changed with the introduction of Sly. Not including them may put Sly on a disadvantage, however, considering that gas giants are more common than other planets types, I think that is a smaller issue than counting gas giants as valid planet types for all other empires, for which they are useless (for colonization).

So I'd remove this particular change.

The proper fix of course would be to consider the species when calculating the merit of a starting position, but that will require a complete rewrite of the homeworld selection algorithm. The current algorithm can't do that at all, because the way it is designed it just chooses a certain requested number of homeworlds.

Assignment from that pool to specific empires happens in a subsequent step, and only then the information what starting species the empire (that gets a homeworld assigned) has is available.

Which means any algorithm that wants to ensure that not only a proper amount of (theoretically) colonizable planets is present near a starting position, but also a proper amount of the right *type* has to be a complete rewrite of the current one, as it needs to consider the starting species. And it will most likely have to be far more complex...

Reply:
> This is potentially very problematic.

In some games I started as Sly, the number of GG at less than 6 hops from my HW was very low (1, 2). For Sly, anything but GG are hostile, which means that not having enough GG around is the most disgraced possible situation (compared to any other species). If a Sly starts with few planets around plus few or no GG, and the algorithm fills the missing planets with anything but GG, the Sly is irremediably doomed. This is very problematic, without the "potentially", and thus deserves some attention.

Although GG are usually more common than any other planetary environment (that is, GG is in general more frequent than Radiated, Terran, etc.), there will be always many more regular planets than GG (i.e. the sum of the nine regular environments is way more frequent than GG), which means that on average regular species will always have more available good+adequate planets than Sly. Except for Narrow Tolerance species, this means than on average regular species always have more early-game colonizable opportunities than Sly.

I assume that the function to add extra planets uses a similar environment frequency than the initial universe generation function, an so the average distribution of planets will stay as more favorable for non-Sly species even if we add GG as an option in that function.

What is worse here? A regular species with few planets around that gets some extra GG (instead of more desirable good/adequate planets) that can be used as a source of extra production with the GG Generator? Or a Sly empire with no good/adequate planets around that get more hostile planets and never a good one?
My impression is that not allowing GG in the function to add extra planets is potentially worse than allowing GG.

> Which means any algorithm that wants to ensure that not only a proper amount of (theoretically) colonizable planets is present near a starting position, but also a proper amount of the right type has to be a complete rewrite of the current one, as it needs to consider the starting species. And it will most likely have to be far more complex...

Totally agree. My initial idea was to reorder the different function in universe generation to accomplish exactly what you say here. The problem is my python skills are close to NULL. This is the first time I do any python programming in my life, so I have to relly on the source code I already have. And so I discarded the more complex solution. However, the problem about unfair starting positions is a already an urging and serious problem. We have players that concede early on in the multiplayer games because their starting positions makes impossible for them to win (and that gives a great advantage for the nearby empires, in detriment of the farther away empires, so the problem cascades), and other players that remain in the game in an effort to not disappoint other players or abandon their team mates, but they stay in a rather discouraged mood which is also bad for everyone else. Some players have stated their decision to not play any more multiplayer games until this is solved. That is serious. Giving an incomplete solution to that issue is arguably better than not giving any solution at all, as long as the partial solution is not making things worse for other cases (but I am rather confident this is not the case).

Summing up, I believe letting that change to stay is better than reverting it. However, I do want your blessing for this change (after you consider the above reasoning), or I'll revert it. At least, the changes are able to improve the situation for most playable species, which is already good to have.

Reply:
I guess you have to run several test games to see how the actual results are, and how much the inclusion of gas giants here affect the presence of usable planets for non-Sly species.

All we can do without testing is trying to make more or less educated guesses, which is probably not sufficient... 😉 

I won't strictly object to keeping your change, if it doesn't work it can always be reverted anyway. But you should keep an eye on it.

Reply:
Otherwise, agreed, we definitely need to revise the placement algorithm to consider species. With the inclusion of species like Sly that became necessary.

Reply:
I'm working on getting it more tested.
Up to know I've started ~10 games with supertesters to see whole galaxy at turn 3 and see all starting positions. Didn't find a "I have many less planets than you" situation yet, but I found several "I have many less good/adequate planets than you". I'll keep an eye on the "I got mostly GG and I'm not Sly, FML" situation, which haven't appeared yet.